engine/interpreter/src/main/scala/pl/touk/nussknacker/engine/compile/ProcessCompiler.scala

```diff
@@ -126,12 +129,23 @@ protected trait ProcessCompilerBase {
     CompilationResult.map3(
       CompilationResult(findDuplicates(splittedProcess.sources).toValidatedNel),
       CompilationResult(nodeTypingInfo, compiledExceptionHandler),
-      splittedProcess.sources.map(compile).sequence
+      compileSources(splittedProcess.sources)
     ) { (_, exceptionHandler, sources) =>
       CompiledProcessParts(splittedProcess.metaData, exceptionHandler, sources)
     }
   }
 
+  private def compileSources(sources: NonEmptyList[SourcePart])(implicit meta: MetaData): CompilationResult[NonEmptyList[PotentiallyStartPart]] = {
+    val zeroAcc = (CompilationResult(Valid(List[PotentiallyStartPart]())), BranchEndContexts(Map(), None))
+    val (result, _) = PartSort.sort(sources.toList).foldLeft(zeroAcc) { case ((resultSoFar, branchContexts), nextPart) =>
```

Comment:
Can we put here some short notice why this sorting is need and reference to longer notice in PartSort? + maybe some TODO if it isn't final solution?

Reply:
Added comment